forge/postoffice/index.js

```diff
@@ -7,68 +7,95 @@ const templates = {}
 module.exports = fp(async function (app, _opts, next) {
     let mailTransport
     let exportableSettings = {}
-
-    // mailTransport = nodemailer.createTransport(require("./localDelivery"));
-
-    // Use a local var as we may decide to disable email if the transport
-    // fails to verify
     let EMAIL_ENABLED = app.config.email.enabled
-
+    let poStartupCheck
+    const isConfigured = EMAIL_ENABLED && (app.config.email.smtp || app.config.email.transport || app.config.email.ses)
     const mailDefaults = { from: app.config.email.from ? app.config.email.from : '"FlowForge Platform" <donotreply@flowforge.com>' }
 
-    if (EMAIL_ENABLED) {
-        if (app.config.email.smtp) {
-            const smtpConfig = app.config.email.smtp
-
-            mailTransport = nodemailer.createTransport(smtpConfig, mailDefaults)
+    // Startup initialisation
+    init(false, (err, connected) => {
```

Comment:
In the case of `app.config.email.transport`, it looks like the callback is called synchronously to this call to `init`. That means `clearTimeout()` will get called *before* the `setInterval` is reached on line 23. So the interval will get setup even though init has been completed. Is that a problem? I'm not sure to be honest - it looks like 5 minutes later the interval will fire, find `EMAIL_ENABLED === true` and call `clearInterval` on itself to stop further checks.
Please have a second look to make sure I've understood that logic correctly.

Reply:
> In the case of app.config.email.transport, it looks like the callback is called synchronously to this call to init

Yes, thats correct.

>  That means clearTimeout() will get called before the setInterval 

The clearInterval call does occur but has zero effect (no exception etc) & in this case, when the interval does occurs, if the transport is unhealthy, the init is called with retry flag - otherwise the clearInterval is called.

It effectively takes care of both the sync and async methods.


> Please have a second look to make sure I've understood that logic correctly.

You got it :)